src/main/java/domain/AttendanceHistories.java

```diff
@@ -0,0 +1,116 @@
+package domain;
+
+import static domain.AttendanceHistory.ABSENT_DEFAULT_HOUR;
+import static domain.AttendanceHistory.ABSENT_DEFAULT_MINUTE;
+import static domain.AttendanceResult.ABSENCE;
+import static domain.AttendanceResult.LATE;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class AttendanceHistories {
+    private final List<AttendanceHistory> histories;
+
+    public AttendanceHistories(List<LocalDateTime> originalHistories, LocalDate standard) {
+        List<LocalDateTime> attendanceTimes = new ArrayList<>(originalHistories);
+        List<LocalDateTime> absenceHistories = generateAbsenceHistories(attendanceTimes, standard);
+        attendanceTimes.addAll(absenceHistories);
+        this.histories = attendanceTimes.stream()
+                .map(AttendanceHistory::new)
+                .collect(Collectors.toList());
+    }
+
+    public AbsenceLevel classifyAbsenceLevel(LocalDateTime standard) {
+        Map<AttendanceResult, Integer> results = getAttendanceResultCount(standard);
+        int absentCount = results.getOrDefault(ABSENCE, 0);
+        int lateCount = results.getOrDefault(LATE, 0);
+        return AbsenceLevel.findAbsenceLevel(absentCount, lateCount);
+    }
+
+    public boolean hasHistory(LocalDateTime time) {
+        return histories.stream()
+                .anyMatch(history -> (history.getAttendanceTime().getDayOfMonth() == time.getDayOfMonth()) &&
+                        (history.getAttendanceTime().getMonthValue() == time.getMonthValue()));
```

Comment:
조금 가독성을 증가시키자면 헬퍼 메서드를 사용해봐도 좋을 것 같아요
getter 사용을 지양하자는 의미도 있구요~
꺼내오지말고 메세지를 보내는것이죠

```suggestion
                .anyMatch(history -> history.isSameDayOfMonth(time) && history.isSameMonth(time);
```

Reply:
AttendanceHistory 내부에 isSameDayOfMonth 메서드와 isSameMonth 메서드를 만들어 리팩토링완료했습니다!